nixos/jenkinsSlave: install java

WolfangAukang opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Generally, when creating a Jenkins agent, it will run a set of steps after obtaining the SSH access to the server. One of these steps is verifying if Java is installed through `java -fullversion`. Currently, if you enable the `services.jenkinsSlave` option, it won't work as it does not have Java installed.

 This change will install the current JDK and will allow the user to specify which package to use.

Added a line on the test to verify the slave has Java installed and this is the snippet that indicates it works:

```
slave: must succeed: java -fullversion
slave # openjdk full version "17.0.1+12-nixos"
(finished: must succeed: java -fullversion, in 0.66 seconds)
```

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

bjornfor commented on nixos/modules/services/continuous-integration/jenkins/slave.nix:
```diff
@@ -46,6 +46,15 @@ in {
         '';
       };
+
+      javaVersion = mkOption {
```
Bikeshed: name it javaPackage instead?

WolfangAukang replied:
Makes sense, just changed it

bjornfor reviewed (commented):
Please explain why this change is needed in the commit message. (Your PR message is very good, just copy it over.)

Tip: if you write the commit message first, your PR automatically gets a copy of the commit message.

WolfangAukang committed:
nixos/jenkinsSlave: install java

Installs Java into the Jenkins agent and allows specifying the JDK/JRE package to use. This is necessary as Jenkins verifies if the agent contains Java installed through the java -fullversion command, which if not, the connection will fail.

WolfangAukang commented:
@bjornfor Ready, I tried to summarize the PR comment into the commit. Let me know if it looks good.

bjornfor reviewed (approved):
Thanks!
